ext/drreg/drreg.c

```diff
@@ -402,7 +402,11 @@ drreg_event_bb_insert_late(void *drcontext, void *tag, instrlist_t *bb, instr_t
         /* Restore aflags to app value */
         LOG(drcontext, LOG_ALL, 3, "%s @%d."PFX" aflags=0x%x: lazily restoring aflags\n",
             __FUNCTION__, pt->live_idx, instr_get_app_pc(inst), aflags);
-        res = drreg_restore_aflags(drcontext, bb, inst, pt, false/*keep slot*/);
+        /* On AArch64, we cannot keep the slot, because spill_reg is used to save the
+         * aflags and this functions requires the slot to be released.
+         */
+        res = drreg_restore_aflags(drcontext, bb, inst, pt,
+                                   IF_AARCHXX_ELSE(true, false/*keep slot*/));
```

Comment:
Wouldn't the complaint be on restore_reg with a different reg, and that assert is already relaxed at https://github.com/DynamoRIO/dynamorio/blob/master/ext/drreg/drreg.c#L207?

If you're seeing an assert on spill_reg, it must be a re-spill after an app write to aflags?  For that, we can relax the spill_reg assert for aflags_slot just like it is relaxed today for restore_reg?